### PR TITLE
chore: add RTL examples for visual testing

### PR DESCRIPTION
**Pull request by mnajdova**
This PR adds mechanism for adding rtl examples for visual test. The added test/examples should end with the extension `.rtl.tsx`, and this is all that is necessary from developer's side (See the example for the `DividerExample.rtl.tsx`). Screener then will maximize this example with rtl enabled and will take screenshot of that.

### CODE REVIEW

File: docs/src/components/ExternalExampleLayout.tsx [hunk 1]
```diff
@@ -11,38 +11,41 @@ import {
...
-        {({ element, error }) => (
-          <>
-            {element}
-            {/* This block allows to see issues with examples as visual regressions. */}
-            {error && <div style={{ fontSize: '5rem', color: 'red' }}>{error.toString()}</div>}
-          </>
-        )}
-      </SourceRender.Consumer>
-    </SourceRender>
+    <Provider theme={{ rtl: rtl === 'true' ? true : false }}>
```
Comment (mnajdova) [inline docs/src/components/ExternalExampleLayout.tsx]: Not sure whether we should wrap this with div that will have dir='rtl' if rtl is enabled. We have that logic in the ComponentExample component.

File: docs/src/components/ExternalExampleLayout.tsx [hunk 2]
```diff
@@ -26,23 +27,29 @@ const ExternalExampleLayout: any = props => {
...
-          <>
-            {element}
-            {/* This block allows to see issues with examples as visual regressions. */}
-            {error && <div style={{ fontSize: '5rem', color: 'red' }}>{error.toString()}</div>}
-          </>
-        )}
-      </SourceRender.Consumer>
-    </SourceRender>
+    <Provider theme={{ rtl: isRtlEnabled }}>
+      {/*<div dir={isRtlEnabled ? 'rtl' : undefined}>*/}
```
Comment (mnajdova) [inline docs/src/components/ExternalExampleLayout.tsx]: Not sure whether we should add this wrapper div that will have dir='rtl' if rtl is enabled. We have that logic in the ComponentExample component.

**Review by alinais — APPROVED** [review]

File: docs/src/examples/components/Divider/Rtl/DividerExample.rtl.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import * as React from 'react'
+import { Divider } from '@stardust-ui/react'
+
+const DividerExampleShorthand = () => <Divider content="مثال النص" />
+
+export default DividerExampleShorthand
```
Comment (layershifter) [inline docs/src/examples/components/Divider/Rtl/DividerExample.rtl.tsx]:
```suggestion
export default DividerExampleRtl
```

May be name should be changed?

File: docs/src/components/ExternalExampleLayout.tsx (hunk 2 above)
Comment (layershifter) [reply docs/src/components/ExternalExampleLayout.tsx]: We definitely need it.

### Without

image


### With

image

File: CHANGELOG.md [hunk 4]
```diff
@@ -20,6 +20,9 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Fixes
 - Fix layout of `Accordion` panel's title @kuzhelov ([#780](https://github.com/stardust-ui/react/pull/780))
 
+### Features
+- Add mechanism for adding rtl visual tests @mnajdova ([#792](https://github.com/stardust-ui/react/pull/792))
```
Comment (layershifter) [inline CHANGELOG.md]: I'm not sure that we should have an entry in `CHANGELOG` about it, there are no any benefits for user from this change

**Review by layershifter — APPROVED** [review]

**Review by miroslavstastny — COMMENTED** [review]
I vote for adding RTL examples to docsite as well (and switch them to RTL by default)

**Comment by mnajdova** [comment]
@miroslavstastny thanks for the suggestion, I've added RTL examples (switch to RTL by default), and now when you maximize an example, if you were seeing the example in RTL the maximize example will also show it in rtl... Please take a look.

File: docs/src/examples/components/Button/index.tsx [hunk 5]
```diff
@@ -13,6 +14,7 @@ const ButtonExamples = () => (
     <Variations />
     <Groups />
     <Usage />
+    <Rtl />
```
Comment (miroslavstastny) [inline docs/src/examples/components/Button/index.tsx:17]: The same change is missing in `Divider/index.tsx`

File: docs/src/routes.tsx [hunk 6]
```diff
@@ -18,7 +18,7 @@ import IntegrateCustomComponents from './views/IntegrateCustomComponents'
 const Router = () => (
   <BrowserRouter basename={__BASENAME__}>
     <Switch>
-      <Route exact path="/maximize/:exampleName" component={ExternalExampleLayout} />
+      <Route exact path="/maximize/:exampleName/:rtl?" component={ExternalExampleLayout} />
```
Comment (miroslavstastny) [inline docs/src/routes.tsx:21]: I don't like `/true` and `/false` in the URL, but let's leave it as is for now and iterate on that later

**Review by miroslavstastny — COMMENTED** [review]
Please also edit `SECTION_ORDER` in `gulp-example-menu.ts` to show RTL below Usage section just above Performance

File: docs/src/examples/components/Button/index.tsx (hunk 5 above)
Comment (mnajdova) [reply docs/src/examples/components/Button/index.tsx:17]: Added, thanks!

**Pull request merged**